rfhub2/cli/population.py

```diff
@@ -0,0 +1,175 @@
+# from pathlib import Path
+import robot.libraries
+from robot.errors import DataError
+from robot.libdocpkg import LibraryDocumentation
+from requests import session, post
+from rfhub2.config import APP_INTERFACE, APP_PORT, BASIC_AUTH_USER, BASIC_AUTH_PASSWORD
+import os
+import re
+
+RESOURCE_PATTERNS = {".robot", ".txt", ".tsv", ".resource"}
+ALL_PATTERNS = (RESOURCE_PATTERNS | {".xml", ".py"})
+PROTOCOL = 'http://'
+API_V1 = 'api/v1'
+
+
+class LibraryPopulation(object):
+
+    def __init__(self, paths, no_installed_keywords):
+        self.paths = paths
+        self.auth = (BASIC_AUTH_USER, BASIC_AUTH_PASSWORD)
+        self.no_installed_keywords = no_installed_keywords
+
+    def add_collections(self) -> None:
+        """
+        Traverses through paths and adds libraries to rfhub.
+        :return:
+        """
+
+        def traverse_paths(path):
+            for item in os.listdir(path):
```

Comment:
Maybe you could switch to using `pathlib` module in general?

Reply:
AFAIK pathlib does not handle listing directories and path.walk.

Reply:
Valid point with walk. Please see `os.scandir` or `Path.iterdir` as substitute for `os.listdir`. `os` functions work with Path objects so you can switch to pathlib and use os functions when necessary. Most use cases in our code should be replaceable and result in more concise code

Reply:
Replaced os.listdir with path.iterdir()